go/libraries/doltcore/sqle/index/prolly_fields.go

```diff
@@ -129,6 +130,68 @@ func GetField(ctx context.Context, td val.TupleDesc, i int, tup val.Tuple, ns tr
 	return v, err
 }
 
+// Serialize writes an interface{} into the byte string representation used in val.Tuple.
+func Serialize(ctx context.Context, ns tree.NodeStore, t val.Type, v interface{}) ([]byte, bool) {
```

Comment:
I think it's worth seeing if we can avoid all this copied code. It seems like we could change this function to something like:
- create new TupleDesc that only has one field of type `t`
- call `NewTupleBuilder` with that new TupleDesc
- set that field in the new TupleBuilder and call `Build`
- return the bytes of that field

I know that's not going to be quite as efficient as this code, but I think it's a win to have less code / less duplicated code, plus I don't think we need to optimize this particular portion of merge where we merge conflicting rows cell-wise together. There are bigger perf issues with merge that would dwarf anything here anyway. 

Plus, it'll avoid us having to jump on firedrills from customers when they get panics because this function doesn't support all the types yet. For example, the first time a customer hits a cell-wise merge conflict resolution with a float, this will panic and we'll have to jump on it. We could fill in all the other data types now of course, but it just seems like a lot of copied code.

Reply:
Good idea. I'll give that a shot.

> For example, the first time a customer hits a cell-wise merge conflict resolution with a float

This isn't currently possible because we only allow specific type changes to be mergible, and those are conversions between string types, set types, and enum types. Presumably as we add support for more types, we would add corresponding tests that will fail until we add more conversions here. So I'm not worried about user panics from unsupported types.

But I like your proposal for implementing this on top of PutField. If it works it could save us some trouble down the line.

Reply:
I think I was initially skeptical because of the inelegance/inefficiency, but I think this was the right approach. Thanks for the suggestion.

Reply:
Sweet! Thanks for trying it out! 😁